### PR TITLE
Emit DiscoveryFetch Event

### PR DESCRIPTION
**Pull request by marcoandredinis**
When a Discovery Service performs a Fetch event, it will emit a DiscoveryFetchEvent.

Each Event contains the cloud provider and the fetcher type. Fetcher type is the resource type as defined in the Matcher. Eg, rds, ec2, aks, gce

### CODE REVIEW

**Comment by GavinFrazar** [comment]
since the ec2 watcher is configured to `getAllAWSServersFetchers` once per minute, this will emit a usage event every minute as well?

File: lib/srv/discovery/discovery.go [hunk 1]
```diff
@@ -1057,6 +1057,40 @@ func (s *Server) emitUsageEvents(events map[string]*usageeventsv1.ResourceCreate
 	return nil
 }
 
+func (s *Server) submitFetchersEvent(fetchers []common.Fetcher) {
+	// Some Matcher types have multiple fetchers.
+	// Eg, rds Matcher converts into `rds` and `aurora` Fetchers
+	// The Fetch Event is meant to represent the configured matcher type: rds
```
Comment (kimlisa) [inline lib/srv/discovery/discovery.go:1063]: i think you meant to finish this sentence 😅

File: lib/srv/discovery/discovery_test.go [hunk 2]
```diff
@@ -114,20 +114,35 @@ func (me *mockEmitter) EmitAuditEvent(ctx context.Context, event events.AuditEve
...
 func (m *mockUsageReporter) AnonymizeAndSubmit(events ...usagereporter.Anonymizable) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.eventsCount++
+	for _, e := range events {
+		switch e.(type) {
+		case *usagereporter.ResourceCreateEvent:
+			m.resourceAddedEventCount++
+		case *usagereporter.DiscoveryFetchEvent:
+			fmt.Printf("EVENT\n %+v\n\n", e)
```
Comment (kimlisa) [inline lib/srv/discovery/discovery_test.go:130]: did you mean to keep this?

**Review by GavinFrazar — APPROVED** [review]

File: lib/srv/discovery/discovery_test.go [hunk 3]
```diff
@@ -1866,6 +1882,8 @@ func TestDiscoveryDatabaseRemovingDiscoveryConfigs(t *testing.T) {
 		t.Fatal("Didn't receive reconcile event after 1s")
 	}
 
+	require.Zero(t, reporter.DiscoveryFetchEventCount(), "a fetch event was emitted but there is not fetchers actually being called")
```
Comment (kimlisa) [inline lib/srv/discovery/discovery_test.go:1885]:
```suggestion
	require.Zero(t, reporter.DiscoveryFetchEventCount(), "a fetch event was emitted but there is no fetchers actually being called")
```

File: lib/srv/discovery/discovery_test.go [hunk 4]
```diff
@@ -1902,6 +1920,8 @@ func TestDiscoveryDatabaseRemovingDiscoveryConfigs(t *testing.T) {
 		case <-time.After(time.Second):
 			t.Fatal("Didn't receive reconcile event after 1s")
 		}
+
+		require.Zero(t, reporter.DiscoveryFetchEventCount(), "a fetch event was emitted but there is not fetchers actually being called")
```
Comment (kimlisa) [inline lib/srv/discovery/discovery_test.go:1924]:
```suggestion
		require.Zero(t, reporter.DiscoveryFetchEventCount(), "a fetch event was emitted but there is no fetchers actually being called")
```

File: lib/srv/discovery/discovery_test.go [hunk 5]
```diff
@@ -1927,6 +1947,7 @@ func TestDiscoveryDatabaseRemovingDiscoveryConfigs(t *testing.T) {
 			return len(srv.dynamicDatabaseFetchers) > 0
 		}, 1*time.Second, 100*time.Millisecond)
 
+		require.Equal(t, 0, reporter.DiscoveryFetchEventCount())
```
Comment (kimlisa) [inline lib/srv/discovery/discovery_test.go:1950]: should this also use `require.Zero` as abvoe?

**Review by kimlisa — APPROVED** [review]

File: lib/srv/discovery/discovery_test.go (hunk 2 above)
Comment (marcoandredinis) [reply lib/srv/discovery/discovery_test.go:130]: 🤦‍♂️
Removed
Thank you

File: lib/srv/discovery/discovery.go (hunk 1 above)
Comment (marcoandredinis) [reply lib/srv/discovery/discovery.go:1063]: It's probably not the best sentence but what I meant is that some Types convert into multiple fetchers, but we only want to capture the type and not the fetcher.

Example
If you add `rds` in the matchers type:
```yaml
aws:
  - types: [1rds]
    regions: [us-east-1]
```
This starts two fetchers: one for RDS and another one for Aurora.
However, we want to emit just a single event for the type, so only `rds`.

I'll try to re-word it

**Pull request merged**